.github/workflows/reusable-test.yaml

```diff
@@ -58,7 +58,7 @@ jobs:
           key: ${{ inputs.os }}-${{ inputs.python-version }}-poetry-${{ hashFiles('poetry.lock') }}
 
       - name: Install python packages
-        run: poetry install
+        run: poetry install --sync
```

Comment:
This is to make sure we remove any packages that poetry doesn't think should be there anymore.